didcomm_messaging/quickstart.py

```diff
@@ -0,0 +1,393 @@
+    # shipping box, the PackagingService will "pack" and "unpack" an encrypted
+    # message. When packing a message, the PackagingService will encrypt the
+    # message to a single target, however. If the message needs to be forwarded
+    # (because the recipient is behind a relay), then those messages will need
+    # to be handled by the RoutingService.
+    packer = PackagingService(resolver, crypto, secrets)
+
+    # The RoutingService handles the routing of messages through relays. When a
+    # message needs to be forwarded, the RoutingService will handle wrapping
+    # each encrypted message within a forward message. The built-in
+    # RoutingService allows for multiple levels of relays, so you don't need to
+    # worry about how a message is routed to the recipient.
+    router = RoutingService(packaging=packer, resolver=resolver)
+
+    # Once everything is setup, we need to store the DID Secrets within the
+    # SecretsService. We do this by taking the secrets that were passed in,
+    # converting them to AskarSecretKey objects (since we are using the
+    # AskarCryptoService), then we associate them with the public key lookup
+    # value that will be sent to us.
+    verkey, xkey = did_secrets
+    await secrets.add_secret(AskarSecretKey(verkey, f"{did}#key-1"))
+    await secrets.add_secret(AskarSecretKey(xkey, f"{did}#key-2"))
+
+    # These can be removed once all keys being sent/received are in #key-N
+    # format. They basically do the same as above, except with the same public
+    # keys that are returned from the CompatibilityPrefixResolver.
+    doc = await resolver.resolve_and_parse(did)
+    await secrets.add_secret(AskarSecretKey(verkey, doc.authentication[0]))
+    await secrets.add_secret(AskarSecretKey(xkey, doc.key_agreement[0]))
+
+    # Finally, we put it all together in the DIDCommMessaging class. The
+    # DIDCommMessaging handles the orchestration of each individual service,
+    # ensuring that messages get packed and delivered via a simple and straight
+    # forward interface.
+    DMP = DIDCommMessaging(
+        crypto=crypto,
+        secrets=secrets,
+        resolver=resolver,
+        packaging=packer,
+        routing=router,
+    )
+
+    return DMP
+
+
+async def send_http_message(
+    dmp: DIDCommMessaging, my_did: DID, message: Message, target: DID
+):
+    """Send a message via HTTP."""
+
+    # Get the message as a dictionary
+    message_wrapper = message
+    message = message_wrapper.as_dict()
+
+    # Encrypt/pack the message to our target
+    packy = await dmp.pack(
+        message=message,
+        to=target,
+        frm=my_did,
+    )
+    packed = packy.message
+
+    # Get the first http endpoint from the last DID in the DID chain
+    endpoint = packy.get_endpoint("http")
+
+    async with aiohttp.ClientSession() as session:
+        LOG.info("posting message type %s to %s" % (message_wrapper.type, endpoint))
```

Comment:
```suggestion
        LOG.info("posting message type %s to %s", message_wrapper.type, endpoint)
```
Need to use this syntax, otherwise the string interpolation will always take place, regardless of log level.

Reply:
Thank you for catching this. Python makes it difficult to keep track with the like, 5+ different ways to print strings, and they don't work across all methods of printing.
```python
print(f"string with {variable} embedded")
print("string with ", variable, " in arg list")
print("string with c-style variables %s" % variable)
log.info(f"string with {variable} embedded")
log.info("string with c-style variables %s" % variable)
log.info("string with c-style variables %s", variable)
```

Reply:
Personally, I prefer the f-string variant followed by the arg-list variant. But I don't believe the latter works in the logger code.

Reply:
lol yeah, it's a problem :sweat_smile: thanks, Python